### PR TITLE
Top-level API for setting use_pytorch_sdpa

### PR DESCRIPTION
**Pull request by pzelasko**
# What does this PR do ?

Top-level API for setting `use_pytorch_sdpa` and selecting the SDPA backend.

**Collection**: ASR

# Changelog 
- Add specific line by line info of high level changes in this PR.

# Usage
* You can potentially add a usage example below

```python
# Add a code snippet demonstrating how to use this 
```

# GitHub Actions CI

The Jenkins CI system has been replaced by GitHub Actions self-hosted runners.

The GitHub Actions CI will run automatically when the "Run CICD" label is added to the PR.
To re-run CI remove and add the label again.
To run CI on an untrusted fork, a NeMo user with write access must first click "Approve and run".

# Before your PR is "Ready for review"
**Pre checks**:
- [ ] Make sure you read and followed [Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md)
- [ ] Did you write any new necessary tests?
- [ ] Did you add or update any necessary documentation?
- [ ] Does the PR affect components that are optional to install? (Ex: Numba, Pynini, Apex etc)
  - [ ] Reviewer: Does the PR have correct import guards for all optional libraries?
  
**PR Type**:
- [ ] New Feature
- [ ] Bugfix
- [ ] Documentation

If you haven't finished some of the above items you can still open "Draft" PR.

## Who can review?

Anyone in the community is free to review the PR once the checks have passed. 
[Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md) contains specific people who can review PRs to various areas.

# Additional Information
* Related to # (issue)

### CODE REVIEW

File: nemo/collections/asr/parts/mixins/mixins.py [hunk 1]
```diff
@@ -556,6 +556,13 @@ def change_attention_model(
                 self.cfg.encoder.self_attention_model = self_attention_model
                 self.cfg.encoder.att_context_size = att_context_size
 
+    def set_use_pytorch_sdpa(self, enabled: bool, sdpa_backends: list[torch.nn.attention.SDPBackend] | None = None):
+        if hasattr(self, "encoder") and hasattr(self.encoder, "layers"):
+            for layer in self.encoder.layers:
+                if hasattr(layer, "self_attn") and hasattr(layer.self_attn, "use_pytorch_sdpa"):
+                    layer.self_attn.use_pytorch_sdpa = enabled
+                    layer.self_attn.use_pytorch_sdpa_backends = sdpa_backends
```
Comment (titu1994) [inline nemo/collections/asr/parts/mixins/mixins.py:564]: Log that the value has been set, or that it could not set it since it could not find the value. Just one log message at the end.

**Review by titu1994 — APPROVED** [review]

File: tests/collections/asr/test_asr_multitask_model_bpe.py [hunk 2]
```diff
@@ -547,6 +547,13 @@ def test_FrameBatchMultiTaskAED(self, asr_model, test_data_dir):
         outputs = model.transcribe()
         assert isinstance(outputs, str)
 
+    @pytest.mark.unit
+    def test_set_use_pytorch_sdpa(self, asr_model):
+        asr_model.set_use_pytorch_sdpa(True)
+        assert asr_model.encoder.layers[0].self_attn.use_pytorch_sdpa == True
```
Comment (orena1) [inline tests/collections/asr/test_asr_multitask_model_bpe.py:553]: maybe also add:
`        assert asr_model.encoder.layers[-1].self_attn.use_pytorch_sdpa == True`
